porter/v1/env_group.proto

```diff
@@ -12,4 +12,27 @@ enum EnumEnvGroupProviderType {
   ENUM_ENV_GROUP_PROVIDER_TYPE_DOPPLER = 2;
   // ENUM_ENV_GROUP_PROVIDER_TYPE_DATASTORE repesents an env group sourced from a managed datastore
   ENUM_ENV_GROUP_PROVIDER_TYPE_DATASTORE = 3;
+  // ENUM_ENV_GROUP_PROVIDER_TYPE_INFISICAL represents an env group sourced from Infisical
+  ENUM_ENV_GROUP_PROVIDER_TYPE_INFISICAL = 4;
+}
+
+// EnumExternalEnvGroupProviderType is the type of external env group provider.
+enum EnumExternalEnvGroupProviderType {
+  // ENUM_EXTERNAL_ENV_GROUP_PROVIDER_TYPE_UNSPECIFIED represents the nil value
+  ENUM_EXTERNAL_ENV_GROUP_PROVIDER_TYPE_UNSPECIFIED = 0;
+  // ENUM_EXTERNAL_ENV_GROUP_PROVIDER_TYPE_EXTERNAL_SECRETS represents the external-secrets env group provider
+  ENUM_EXTERNAL_ENV_GROUP_PROVIDER_TYPE_EXTERNAL_SECRETS = 1;
+  // ENUM_EXTERNAL_ENV_GROUP_PROVIDER_TYPE_INFISICAL represents the Infisical env group provider
+  ENUM_EXTERNAL_ENV_GROUP_PROVIDER_TYPE_INFISICAL = 2;
+}
```

Comment:
maybe we call this the EnvGroupDriver or EnvGroupOperator or EnvGroupIntegrator? Just to distinguish more from the ProviderType (`external-secrets` operator does not provide, it just helps us integrate with most providers)

Reply:
yeah fair, EnvGroupOperator makes sense to me